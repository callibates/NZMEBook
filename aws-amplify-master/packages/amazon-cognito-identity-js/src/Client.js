import UserAgent from './UserAgent';
/** @class */
export default class Client {
  /**
   * Constructs a new AWS Cognito Identity Provider client object
   * @param {string} region AWS region
   * @param {string} endpoint endpoint
   */
  constructor(region, endpoint) {
    this.endpoint = endpoint || `https://cognito-idp.${region}.amazonaws.com/`;
    this.userAgent = UserAgent.prototype.userAgent || 'aws-amplify/0.1.x js';
  }

  /**
   * Makes an unauthenticated request on AWS Cognito Identity Provider API
   * using fetch
   * @param {string} operation API operation
   * @param {object} params Input parameters
   * @param {function} callback Callback called when a response is returned
   * @returns {void}
  */
  request(operation, params, callback) {
    const headers = {
      'Content-Type': 'application/x-amz-json-1.1',
      'X-Amz-Target': `AWSCognitoIdentityProviderService.${operation}`,
      'X-Amz-User-Agent': this.userAgent,
    };

    const options = {
      headers,
      method: 'POST',
      mode: 'cors',
      cache: 'no-cache',
      body: JSON.stringify(params),
    };

    let response;
    let responseJsonData;

    fetch(this.endpoint, options)
      .then(resp => {
        response = resp;
        return resp;
      })
      .catch(err => {
        // If error happens here, the request failed
        // if it is TypeError throw network error
        if (err instanceof TypeError) {
          throw new Error('Network error');
        }
        throw err;
      })
      .then(resp => resp.json())
      .catch(err => {
        // If error happens here
        // cannot parse the body stream, return undefined
        if (response.ok) return callback(null, undefined);
        else {
          const error = {
            code: response.status, 
            statusCode: response.status,
            message: response.statusText
          }
          callback(error);
        }
      })
      .then(data => {
        // return parsed body stream
        if (response.ok) return callback(null, data);
        responseJsonData = data;

        // Taken from aws-sdk-js/lib/protocol/json.js
        // eslint-disable-next-line no-underscore-dangle
        const code = (data.__type || data.code).split('#').pop();
        const error = {
          code,
          name: code,
          message: (data.message || data.Message || null),
        };
        return callback(error);
      })
      .catch(err => {
        // if cannot split the data
        // default to return 'UnknownError' with the json data from response
        let error = { code: 'UnknownError', message: 'Unknown error, the response body from fetch is: ' + responseJsonData};

        // first check if we have a service error
        if (response && response.headers && response.headers.get('x-amzn-errortype')) {
          try {
            const code = (response.headers.get('x-amzn-errortype')).split(':')[0];
            error = {
              code,
              name: code,
              statusCode: response.status,
              message: (response.status) ? response.status.toString() : null,
            };
          } catch (ex) {
              // pass through so it doesn't get swallowed if we can't parse it
              error = {
                code: 'UnknownError',
                message: response.headers.get('x-amzn-errortype'),
              }
              return callback(error);
          }
        // otherwise check if error is Network error
        } else if (err instanceof Error && err.message === 'Network error') {
          error = {
            code: 'NetworkError',
            name: err.name,
            message: err.message,
          };
        }
        return callback(error);
      });
  }
}
