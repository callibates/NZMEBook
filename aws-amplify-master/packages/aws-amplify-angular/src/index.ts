export { AmplifyAngularModule } from './aws-amplify-angular.module';

export * from './components';
export * from './providers';
