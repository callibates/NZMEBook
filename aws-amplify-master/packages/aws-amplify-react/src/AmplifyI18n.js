/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const dict = {
    'fr': {
        'Sign In': "Se connecter",
        'Sign Up': "S'inscrire",
        'Sign Out': "Déconnexion",
        'Forgot Password': "Mot de passe oublié",
        'Username': "Nom d'utilisateur",
        'Password': "Mot de passe",
        'Change Password': "Changer le mot de passe",
        'New Password': "nouveau mot de passe",
        'Email': "Email",
        'Phone Number': "Numéro de téléphone",
        'Confirm a Code': "Confirmer un code",
        'Confirm Sign In': "Confirmer la connexion",
        'Confirm Sign Up': "Confirmer l'inscription",
        'Back to Sign In': "Retour à la connexion",
        'Send Code': "Envoyer le code",
        'Confirm': "Confirmer",
        'Resend a Code': "Renvoyer un code",
        'Submit': "Soumettre",
        'Skip': "Sauter",
        'Verify': "Vérifier",
        'Verify Contact': "Vérifier le contact",
        'Code': "Code",
        'Account recovery requires verified contact information':
"La récupération du compte nécessite des informations de contact vérifiées",

        'User does not exist': "L'utilisateur n'existe pas",
        'User already exists': "L'utilisateur existe déjà",
        'Incorrect username or password': "identifiant ou mot de passe incorrect",
        'Invalid password format': "format de mot de passe invalide",
        'Invalid phone number format':
`Format de numéro de téléphone invalide.
Veuillez utiliser un format de numéro de téléphone du +12345678900`
    },

    'es': {
        'Sign In': "Registrarse",
        'Sign Up': "Regístrate",
        'Sign Out': "Desconectar",
        'Forgot Password': "Se te olvidó tu contraseña",
        'Username': "Nombre de usuario",
        'Password': "Contraseña",
        'Change Password': "Cambia la contraseña",
        'New Password': "Nueva contraseña",
        'Email': "Email",
        'Phone Number': "Número de teléfono",
        'Confirm a Code': "Confirmar un código",
        'Confirm Sign In': "Confirmar inicio de sesión",
        'Confirm Sign Up': "Confirmar Registrarse",
        'Back to Sign In': "Volver a Iniciar sesión",
        'Send Code': "Enviar código",
        'Confirm': "Confirmar",
        'Resend a Code': "Reenviar un código",
        'Submit': "Enviar",
        'Skip': "Omitir",
        'Verify': "Verificar",
        'Verify Contact': "Verificar contacto",
        'Code': "Código",
        'Account recovery requires verified contact information':
"La recuperación de la cuenta requiere información de contacto verificada",

        'User does not exist': "el usuario no existe",
        'User already exists': "El usuario ya existe",
        'Incorrect username or password': "Nombre de usuario o contraseña incorrecta",
        'Invalid password format': "Formato de contraseña inválido",
        'Invalid phone number format':
`Formato de número de teléfono inválido.
Utilice el formato de número de teléfono +12345678900`
    },

    'zh': {
        'Sign In': "登录",
        'Sign Up': "注册",
        'Sign Out': "退出",
        'Forgot Password': "忘记密码",
        'Username': "用户名",
        'Password': "密码",
        'Change Password': "改变密码",
        'New Password': "新密码",
        'Email': "邮箱",
        'Phone Number': "电话",
        'Confirm a Code': "确认码",
        'Confirm Sign In': "确认登录",
        'Confirm Sign Up': "确认注册",
        'Back to Sign In': "回到登录",
        'Send Code': "发送确认码",
        'Confirm': "确认",
        'Resend a Code': "重发确认码",
        'Submit': "提交",
        'Skip': "跳过",
        'Verify': "验证",
        'Verify Contact': "验证联系方式",
        'Code': "确认码",
        'Account recovery requires verified contact information':
"账户恢复需要验证过的联系方式",

        'User does not exist': "用户不存在",
        'User already exists': "用户已经存在",
        'Incorrect username or password': "用户名或密码错误",
        'Invalid password format': "密码格式错误",
        'Invalid phone number format': "电话格式错误，请使用格式 +12345678900"
    }
}

export default dict;
