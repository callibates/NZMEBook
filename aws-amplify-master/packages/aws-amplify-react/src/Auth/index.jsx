/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Authenticator from './Authenticator';

export { default as Authenticator } from './Authenticator';
export { default as AuthPiece } from './AuthPiece';
export { default as SignIn } from './SignIn';
export { default as ConfirmSignIn } from './ConfirmSignIn';
export { default as SignOut } from './SignOut';
export { default as RequireNewPassword } from './RequireNewPassword';
export { default as SignUp } from './SignUp';
export { default as ConfirmSignUp } from './ConfirmSignUp';
export { default as VerifyContact } from './VerifyContact';
export { default as ForgotPassword } from './ForgotPassword';
export { default as Greetings } from './Greetings';
export { default as FederatedSignIn, FederatedButtons } from './FederatedSignIn';
export { default as TOTPSetup } from './TOTPSetup';

export * from './Provider';

import Greetings from './Greetings';

export function withAuthenticator(Comp, includeGreetings=false, authenticatorComponents = []) {
    return class extends Component {
        constructor(props) {
            super(props);

            this.handleAuthStateChange = this.handleAuthStateChange.bind(this);

            this.state = {
                authState: props.authState || null,
                authData: props.authData || null
            };
        }

        handleAuthStateChange(state, data) {
            this.setState({ authState: state, authData: data });
        }

        render() {
            const { authState, authData } = this.state;
            const signedIn = (authState === 'signedIn');
            if (signedIn) {
                return (
                    <div>
                        {
                            includeGreetings?
                            <Greetings
                                authState={authState}
                                authData={authData}
                                onStateChange={this.handleAuthStateChange}
                            />
                            : null
                        }
                        <Comp
                            {...this.props}
                            authState={authState}
                            authData={authData}
                            onStateChange={this.handleAuthStateChange}
                        />
                    </div>
                )
            }

            return <Authenticator
                {...this.props}
                hideDefault={authenticatorComponents.length > 0}
                onStateChange={this.handleAuthStateChange}
                children={authenticatorComponents}
            />
        }
    }
}

export class AuthenticatorWrapper extends Component {
    constructor(props) {
        super(props);

        this.state = { auth: 'init' };

        this.handleAuthState = this.handleAuthState.bind(this);
    }

    handleAuthState(state, data) {
        this.setState({ auth: state, authData: data });
    }

    render() {
        return (
            <div>
                <Authenticator {...this.props} onStateChange={this.handleAuthState} />
                {this.props.children(this.state.auth)}
            </div>
        )
    }
}

AuthenticatorWrapper.propTypes = {
    children: PropTypes.func.isRequired
}
