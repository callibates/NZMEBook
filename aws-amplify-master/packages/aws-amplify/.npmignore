__mocks__/**
__tests__/**
coverage/**
docs/**
node_modules/**
src/**
.vscode/**
.DS_Store
*.log
prepend-license.js
prettier.config.json
tsconfig.json
tsfmt.json
tslint.json
typeDoc.js
webpack.config.js