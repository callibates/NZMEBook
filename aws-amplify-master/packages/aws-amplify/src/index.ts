/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import Analytics, { AnalyticsClass, AnalyticsProvider, AWSPinpointProvider, AWSKinesisProvider } from './Analytics';
import Auth, { AuthClass } from './Auth';
import Storage, { StorageClass } from './Storage';
import API, { APIClass, graphqlOperation } from './API';
import PubSub, { PubSubClass } from './PubSub';
import Cache from './Cache';
import Interactions, { InteractionsClass } from './Interactions';
import {
    ConsoleLogger as Logger,
    Hub,
    JS,
    ClientDevice,
    Signer,
    I18n,
    Amplify,
    ServiceWorker
} from './Common';

export default Amplify;

Amplify.Auth = Auth;
Amplify.Analytics = Analytics;
Amplify.API = API;
Amplify.Storage = Storage;
Amplify.I18n = I18n;
Amplify.Cache = Cache;
Amplify.PubSub = PubSub;
Amplify.Logger = Logger;
Amplify.ServiceWorker = ServiceWorker;
Amplify.Interactions = Interactions;

export { 
    Auth, 
    Analytics, 
    Storage,
    API, 
    PubSub, 
    I18n, 
    Logger, 
    Hub, 
    Cache, 
    JS, 
    ClientDevice, 
    Signer, 
    ServiceWorker, 
    Interactions 
};

export { 
    AuthClass, 
    AnalyticsClass, 
    APIClass, 
    StorageClass,
    PubSubClass,
    InteractionsClass,
    AnalyticsProvider, 
    AWSPinpointProvider, 
    AWSKinesisProvider 
};
export { graphqlOperation };
