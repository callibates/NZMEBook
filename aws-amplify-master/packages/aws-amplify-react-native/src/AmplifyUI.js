/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import React from 'react';
import { View, Text, TextInput, TouchableHighlight } from 'react-native';

import { I18n } from 'aws-amplify';

export const Username = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Username')}
            autoFocus={true}
            autoCapitalize="none"
            {...props}
        />
    )
}

export const Password = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Password')}
            secureTextEntry={true}
            {...props}
        />
    )
}

export const Email = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Email')}
            keyboardType="email-address"
            autoCapitalize="none"
            {...props}
        />
    )
}

export const PhoneNumber = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Phone Number')}
            keyboardType="phone-pad"
            {...props}
        />
    )
}

export const ConfirmationCode = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <TextInput
            style={theme.input}
            placeholder={I18n.get('Code')}
            autoFocus={true}
            {...props}
        />
    )
}

export const LinkCell = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.cell}>
            <TouchableHighlight
                onPress={props.onPress}
            >
                <Text style={theme.sectionFooterLink}>{props.children}</Text>
            </TouchableHighlight>
        </View>
    )
}

export const Header = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.sectionHeader}>
            <Text style={theme.sectionHeaderText}>{props.children}</Text>
        </View>
    )
}

export const ErrorRow = (props) => {
    const theme = props.theme || AmplifyTheme;
    return (
        <View style={theme.errorRow}>
            <Text style={theme.erroRowText}>{props.children}</Text>
        </View>
    )
}
