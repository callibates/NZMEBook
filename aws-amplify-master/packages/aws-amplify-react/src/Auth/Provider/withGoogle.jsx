import React, { Component } from 'react';

import { Auth, Logger } from 'aws-amplify';
import AmplifyTheme from '../../AmplifyTheme';
import { SignInButton } from '../../AmplifyUI';

const logger = new Logger('withGoogle');

export default function withGoogle(Comp) {
    return class extends Component {
        constructor(props) {
            super(props);

            this.initGapi = this.initGapi.bind(this);
            this.signIn = this.signIn.bind(this);
            this.federatedSignIn = this.federatedSignIn.bind(this);

            this.state = {};
        }

        signIn() {
            const ga = window.gapi.auth2.getAuthInstance();
            const { onError } = this.props;
            ga.signIn().then(
                googleUser => this.federatedSignIn(googleUser),
                error => {
                    if (onError) onError(error);
                    else throw error;
                }
            );
        }

        federatedSignIn(googleUser) {
            const { id_token, expires_at } = googleUser.getAuthResponse();
            const profile = googleUser.getBasicProfile();
            const user = {
                email: profile.getEmail(),
                name: profile.getName()
            };

            const { onStateChange } = this.props;
            return Auth.federatedSignIn(
                'google',
                { token: id_token, expires_at },
                user
            ).then(credentials => {
                if (onStateChange) {
                    onStateChange('signedIn');
                }
            });
        }

        componentDidMount() {
            const { google_client_id } = this.props;
            if (google_client_id) this.createScript();
        }

        createScript() {
            const script = document.createElement('script');
            script.src = 'https://apis.google.com/js/platform.js';
            script.async = true;
            script.onload = this.initGapi;
            document.body.appendChild(script);
        }

        initGapi() {
            logger.debug('init gapi');

            const that = this;
            const { google_client_id } = this.props;
            const g = window.gapi;
            g.load('auth2', function() {
                g.auth2.init({
                    client_id: google_client_id,
                    scope: 'profile email openid'
                });
            });
        }

        render() {
            const ga =
                window.gapi && window.gapi.auth2
                    ? window.gapi.auth2.getAuthInstance()
                    : null;
            return <Comp {...this.props} ga={ga} googleSignIn={this.signIn} />;
        }
    };
}

const Button = props => (
    <SignInButton
        id="google_signin_btn"
        onClick={props.googleSignIn}
        theme={props.theme || AmplifyTheme}
    >
        Sign In with Google
    </SignInButton>
);

export const GoogleButton = withGoogle(Button);
