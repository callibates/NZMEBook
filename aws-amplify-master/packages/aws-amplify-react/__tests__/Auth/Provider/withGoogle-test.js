import React, { Component } from 'react';
import withGoogle, {
    GoogleButton
} from '../../../src/Auth/Provider/withGoogle';
import { SignInButton, Button } from '../../../src/AmplifyUI';
import { Auth } from 'aws-amplify';

describe('withGoogle test', () => {
    describe('render test', () => {
        test('render correctly', () => {
            window.gapi = null;
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };
            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('signIn test', () => {
        test('happy case with connected response', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        return {
                            signIn() {
                                return new Promise((res, rej) => {
                                    res('googleUser');
                                });
                            }
                        };
                    }
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            const comp = wrapper.instance();

            const spyon = jest
                .spyOn(comp, 'federatedSignIn')
                .mockImplementationOnce(() => {
                    return;
                });

            await comp.signIn();

            expect(spyon).toBeCalledWith('googleUser');

            spyon.mockClear();
        });

        test('with an onError handler', async () => {
            const onErrorMock = jest.fn();

            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        return {
                            signIn() {
                                return {
                                    then: (success, error) => error('error')
                                };
                            }
                        };
                    }
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp onError={onErrorMock} />);
            const comp = wrapper.instance();

            comp.signIn();

            expect(onErrorMock).toBeCalledWith('error');

            onErrorMock.mockClear();
        });

        test('without an onError handler', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        return {
                            signIn() {
                                return {
                                    then: (success, error) => error('error')
                                };
                            }
                        };
                    }
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            const comp = wrapper.instance();

            expect(() => comp.signIn()).toThrowError('error');
        });
    });

    describe('federatedSignIn', () => {
        test('happy case', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            const googleUser = {
                getAuthResponse() {
                    return {
                        id_token: 'id_token',
                        expires_at: 0
                    };
                },
                getBasicProfile() {
                    return {
                        getEmail() {
                            return 'email';
                        },
                        getName() {
                            return 'name';
                        }
                    };
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            const comp = wrapper.instance();

            const spyon = jest
                .spyOn(Auth, 'federatedSignIn')
                .mockImplementationOnce(() => {
                    return new Promise((res, rej) => {
                        res('credentials');
                    });
                });

            await comp.federatedSignIn(googleUser);

            expect(spyon).toBeCalledWith(
                'google',
                { expires_at: 0, token: 'id_token' },
                { email: 'email', name: 'name' }
            );

            spyon.mockClear();
        });

        test('happy case with onStateChange exists', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            const mockFn = jest.fn();

            const googleUser = {
                getAuthResponse() {
                    return {
                        id_token: 'id_token',
                        expires_at: 0
                    };
                },
                getBasicProfile() {
                    return {
                        getEmail() {
                            return 'email';
                        },
                        getName() {
                            return 'name';
                        }
                    };
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            const comp = wrapper.instance();
            wrapper.setProps({
                onStateChange: mockFn
            });

            const spyon = jest
                .spyOn(Auth, 'federatedSignIn')
                .mockImplementationOnce(() => {
                    return new Promise((res, rej) => {
                        res('credentials');
                    });
                });

            await comp.federatedSignIn(googleUser);

            expect(spyon).toBeCalledWith(
                'google',
                { expires_at: 0, token: 'id_token' },
                { email: 'email', name: 'name' }
            );
            expect(mockFn).toBeCalledWith('signedIn');

            spyon.mockClear();
        });
    });

    describe('initGapi test', () => {
        test('happy case', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            };

            window.gapi = {
                load(path, callback) {
                    callback();
                },
                auth2: {
                    init(params) {
                        return new Promise((res, rej) => {
                            res('ga');
                        });
                    },
                    getAuthInstance() {
                        return {
                            signIn() {
                                return new Promise((res, rej) => {
                                    res('googleUser');
                                });
                            }
                        };
                    }
                }
            };

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp />);
            const comp = wrapper.instance();

            await comp.initGapi();
        });
    });

    describe.skip('refreshGoogleToken test', () => {
        test('happy case', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }

            const authResponse = {
                id_token: 'id_token',
                expires_at: 0
            }

            const googleAuth = {
                currentUser: {
                    get() {
                        // google User
                        return {
                            isSignedIn() {
                                return true;
                            },
                            reloadAuthResponse() {
                                return new Promise((res, rej) => {
                                    res(authResponse);
                                });
                            },
                            getBasicProfile() {
                                return {
                                    getEmail() { return 'email' },
                                    getName() { return 'name' }
                                }
                            }
                        }
                    }
                }
            }

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        // googleAuth
                        return new Promise((res, rej) => {
                            res(googleAuth);
                        });
                    }
                }
            }

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp/>);
            const comp = wrapper.instance();

            const spyon = jest.spyOn(Auth, 'federatedSignIn').mockImplementationOnce(() => { return Promise.resolve() });

            await comp.refreshGoogleToken();

            spyon.mockClear();
            
        });

        test('not signed in', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }

            const authResponse = {
                id_token: 'id_token',
                expires_at: 0
            }

            const googleAuth = {
                currentUser: {
                    get() {
                        // google User
                        return {
                            isSignedIn() {
                                return false;
                            }
                        }
                    }
                }
            }

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        // googleAuth
                        return new Promise((res, rej) => {
                            res(googleAuth);
                        });
                    }
                }
            }

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp/>);
            const comp = wrapper.instance();
            await comp.refreshGoogleToken();
            
        });

        test('no auth2', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }
            window.gapi = null

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp/>);
            const comp = wrapper.instance();

            await comp.refreshGoogleToken();
        });

        test('no googleAuth instance', async () => {
            const MockComp = class extends Component {
                render() {
                    return <div />;
                }
            }
            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        // googleAuth
                        return new Promise((res, rej) => {
                            res(null);
                        });
                    }
                }
            }

            const Comp = withGoogle(MockComp);
            const wrapper = shallow(<Comp/>);
            const comp = wrapper.instance();

            await comp.refreshGoogleToken();
        });
    });
});

describe('GoogleButton test', () => {
    describe('render test', () => {
        test('render correctly', () => {
            window.gapi = null;
            const wrapper = shallow(<GoogleButton />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
