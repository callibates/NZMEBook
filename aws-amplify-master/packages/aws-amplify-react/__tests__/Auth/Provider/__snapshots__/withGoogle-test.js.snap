// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleButton test render test render correctly 1`] = `
<Button
  ga={null}
  googleSignIn={[Function]}
/>
`;

exports[`withGoogle test render test render correctly 1`] = `
<MockComp
  ga={null}
  googleSignIn={[Function]}
/>
`;
